src/components/NavMenu.tsx

```diff
@@ -2,72 +2,133 @@ import React, { useState, Fragment } from "react";
 import { NavLink } from "react-router-dom";
 import styled, { css } from "styled-components";
 import { useTranslation } from "react-i18next";
-import navButton from "./menu-button.png";
+import menuButton from "./assets/nav-menu/menu-button.png";
 import {
+  color,
+  layout,
   space,
-  SpaceProps,
-  TypographyProps,
   typography,
-  layout,
-  LayoutProps,
   position,
-  PositionProps,
-  color,
-  ColorProps,
   border,
+  background,
+  grid,
+  ColorProps,
+  LayoutProps,
+  SpaceProps,
+  TypographyProps,
+  PositionProps,
   BorderProps,
+  BackgroundProps,
+  GridProps,
 } from "styled-system";
 
-const visibleStyles = css`
+const overlayStyles = css`
   display: flex;
   flex-direction: column;
   height: 100%;
-  background-color: rgba(0, 0, 0, 0.1);
-  font-size: 36px;
-  justify-content: center;
-  text-align: center;
   width: 100%;
-  font-family: SangBleu OG Serif Light Regular;
+  justify-content: center;
   border: none;
+  background-image: url("/assets/nav-menu/background-inverted.png");
+  background-size: cover;
+  opacity: 1;
 `;
-const NavMenuOverlay = styled.dialog<{ isOpen: boolean }>`
+
+const Overlay = styled.dialog<{ isOpen: boolean }>`
   display: none;
-  ${(props) => props.isOpen && visibleStyles}
+  ${(props) => props.isOpen && overlayStyles}
 `;
 
-type MenuProps = SpaceProps &
-  TypographyProps &
-  PositionProps &
-  LayoutProps &
-  ColorProps &
-  BorderProps;
+const Logo = styled.p<PositionProps & LayoutProps & TypographyProps>`
+${position}
+${layout}
+${typography}
+`;
 
-const MenuButton = styled.button<MenuProps>`
-  ${position}
-  ${layout}
-  ${border}
-  ${color}
+const MenuButton = styled.button<
+  PositionProps & LayoutProps & BorderProps & BackgroundProps & TypographyProps
+>`
+${position}
+${layout}
+${border}
+${background}
+${typography}
 `;
 
-const MenuItem = styled.li<MenuProps>`
-  ${space}
-  ${typography}
-  ${layout}
-  ${color}
-  &:focus,
-  &:hover {
-    font-size: 32px;
+const Grid = styled.div<GridProps>`
+  display: grid;
+  ${grid};
+`;
+
+const MenuContainer = styled.div<SpaceProps>`
+  ${space};
+`;
+
+const Menu = styled.ul<GridProps & TypographyProps>`
+  ${grid};
+  ${typography};
+`;
+
+const MenuLink = styled(NavLink)<ColorProps & SpaceProps & TypographyProps>`
+  ${color};
+  ${space};
+  ${typography};
   }
-  &:visited,
-  &:link,
-  &:active {
-    text-decoration: none;
+`;
+
+const MenuText = styled.li`
+  &:hover {
+    transition: transform 0.2s;
+    transform: scale(1.01);
```

Comment:
Sooo this is a good spot because it flagged a knowledge gap on the use of `justify-content` for me. I used it to line everything up, which indeed it does but there's obviously the implication of spreading out the text across its parent. Anyway, I've swapped it for `textAlign: start` which I think it more suitable here.